src/test/groovy/graphql/analysis/MaxBatchOperationsInstrumentationTest.groovy

```diff
@@ -0,0 +1,163 @@
+        def executeOperationParameters = new InstrumentationExecuteOperationParameters(executionContext)
+        def state = maxBatchOperationsInstrumentation.createState(null)
+        when:
+        maxBatchOperationsInstrumentation.beginExecuteOperation(executeOperationParameters, state)
+        then:
+        notThrown(Exception)
+    }
+
+    def "doesn't throw exception if number of operations are below maximum with deprecated beginExecuteOperation"() {
+        given:
+        def schema = TestUtil.schema("""
+            type Query{
+                foo: Foo
+                bar: String
+            }
+            type Foo {
+                scalar: String  
+                foo: Foo
+            }
+        """)
+        def query = createQuery("""
+            {f1: foo {foo {foo {scalar}}} f2: foo { foo {foo {foo {foo{foo{scalar}}}}}} }
+            """)
+        MaxBatchOperationsInstrumentation maxBatchOperationsInstrumentation = new MaxBatchOperationsInstrumentation(7)
+        ExecutionInput executionInput = Mock(ExecutionInput)
+        def executionContext = executionCtx(executionInput, query, schema)
+        def executeOperationParameters = new InstrumentationExecuteOperationParameters(executionContext)
+        when:
+        maxBatchOperationsInstrumentation.beginExecuteOperation(executeOperationParameters, null) // Retain for test coverage
+        then:
+        notThrown(Exception)
+    }
+
+    def "custom max batch operation exceeded function"() {
+        given:
+        def schema = TestUtil.schema("""
+            type Query{
+                foo: Foo
+                bar: String
+            }
+            type Foo {
+                scalar: String  
+                foo: Foo
+            }
+        """)
+        def query = createQuery("""
+            {f1: foo {foo {foo {scalar}}} f2: foo { foo {foo {foo {foo{foo{scalar}}}}}} f3: foo {scalar} f4: foo {scalar} }
+            """)
+        Boolean calledFunction = false
+        Function<RequestWidthInfo, Boolean> maxBatchOperationsExceededFunction = new Function<RequestWidthInfo, Boolean>() {
+            @Override
+            Boolean apply(final RequestWidthInfo requestWidthInfo) {
+                calledFunction = true
+                return false
+            }
+        }
+        MaxBatchOperationsInstrumentation maxBatchOperationsInstrumentation = new MaxBatchOperationsInstrumentation(2, maxBatchOperationsExceededFunction)
+        ExecutionInput executionInput = Mock(ExecutionInput)
+        def executionContext = executionCtx(executionInput, query, schema)
+        def executeOperationParameters = new InstrumentationExecuteOperationParameters(executionContext)
+        when:
+        maxBatchOperationsInstrumentation.beginExecuteOperation(executeOperationParameters, null)
+        then:
+        calledFunction
+        notThrown(Exception)
+    }
+
+    def "coercing null variables that are marked as non nullable wont blow up early"() {
```

Comment:
Whats this test got to do with the code??  That validation works?

Your instrumentation kicks off at `beginExecution` its a garuteeed position

Reply:
Since this use case was similar to max query depth instrumentation, i tried to keep everything including the tests similar to it.

As i understand it, this test is verifying that if the input validation fails then it should return errors at the correct place(validation should fail), and the instrumentation code should not interfere with it.